components/ParticipantFormUpdate.tsx

```diff
@@ -29,10 +30,15 @@ export default function ParticipantFormUpdate({
 
   const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    await ParticipantService(eventId).update(participantData.id, data);
-    setIsSubmitted(true);
+
+    try {
+      await ParticipantService(eventId).update(participantData.id, data);
+      setIsSubmitted(true);
+    } catch (error) {
+      setIsFailed(true);
+    }
```

Comment:
I see this as part of the setTimeout block, as you want to redirect the user after a certain time and not a certain time after the request, correct?

Reply:
yessir!